### ISSUE TITLE
MAM transformation task causes build failure (intuneDebugMamifyClasses/classes.jar already contains entry)

### ISSUE THREAD

**Issue by parveen-bhatia** [msintuneappsdk/ms-intune-app-sdk-android#222]
# Intune Android App SDK [Gradle Build Plugin] Issue

## Questions to Ask Before Submission

1. Does your app compile and launch successfully without the Intune App SDK? - Yes
2. Have you checked the [Microsoft Intune App SDK for Android] repository for similar issues? - Yes
3. Are you using the latest version of the [SDK and Plugin]? Yes

## Summary

Please briefly describe the issue you are experiencing.

After upgrading our's project to AGP/Gradle 8.x, we upgraded Intune SDK and MAM plugin from 9.5.0 to 10.0.0. After that we started seeing this build failure:

`Task :appian-android:intuneDebugMamifyClasses FAILED`

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':appian-android:intuneDebugMamifyClasses'.
> Zip file '/Users/parveen.bhatia/repo/android-client/appian-android/build/intermediates/classes/intuneDebug/ALL/intuneDebugMamifyClasses/classes.jar' already contains entry 'com/appiancorp/core/expr/fn/modules/core/ClientEvaluatorOverridenFunctions.class', cannot overwrite
```

`ClientEvaluatorOverridenFunctions.class` is the part of our's internal libaray (.aar) which is included in our's project.
As this library doesn't interact with system API's. We tried to exclude this class as suggested in [documentation](https://learn.microsoft.com/en-us/mem/intune/developer/app-sdk-android-phase3#exclusions) using following, but still getting same issue.

```
intunemam {
excludeClasses: ['com.appiancorp.core.expr.fn.modules.core.ClientEvaluatorOverridenFunctions']
} 
```

Questions:
- How we can exclude this .aar library from the mam transformations ?
- Is there any other way to fix this problem apart from exclusions ?

## Details

- **Intune Android App SDK Version:** 10.0.0
- **Gradle Version:**  8.5
- **AGP Version:** 8.1.1
- **Operating System (Mac, Linux, Windows) and Version:** MacOS 14.3.1
- **Android Studio Version:** Android Studio Hedgehog | 2023.1.1 Patch 2

### Android Studio / Gradle Logs
[gradle_build_failure.txt](https://github.com/msintuneappsdk/ms-intune-app-sdk-android/files/14670872/gradle_build_failure.txt)

### Gradle Build Plugin Report Logs

[manifest-merger-intune-debug-report.txt.zip](https://github.com/msintuneappsdk/ms-intune-app-sdk-android/files/14670904/manifest-merger-intune-debug-report.txt.zip)

**Comment by mukeshk-ms** [comment]
@parveen-bhatia can you please refer to this [issue](https://github.com/msintuneappsdk/ms-intune-app-sdk-android/issues/186) and see if the suggested solution helps in resolving it

**Comment by parveen-bhatia** [comment]
@mukeshk-ms i was able to fix this issue after upgrading AGP from 8.1.1 to 8.3.1.
But i think it would be great if Intune SDK release notes and documentation can be updated about compatible AGP and Gradle versions, so that these kind of issues can be avoided. Thanks.
